feat(docs): add eas hosting option (#34150)
expo/expo: docs/pages/router/reference/api-routes.mdx

## Deployment

> When you're ready to deploy to production, simply run [`npx expo export`](/more/expo-cli#exporting) to create the server bundle in the `dist` folder. This server can be tested locally with `npx expo serve` (Expo SDK 52), visit the URL in a web browser or create a native build with the `origin` set to the local server URL.

Reviewer: ```suggestion
When you're ready to deploy to production, run [`npx expo export`](/more/expo-cli#exporting) to create the server bundle in the `dist` folder. This server can be tested locally with `npx expo serve` (available in Expo SDK 52 and higher), visit the URL in a web browser or create a native build with the `origin` set to the local server URL.
```

> When you're ready to deploy to production, run [`npx expo export`](/more/expo-cli#exporting) to create the server bundle in the `dist` folder. This server can be tested locally with `npx expo serve` (available in Expo SDK 52 and higher), visit the URL in a web browser or create a native build with the `origin` set to the local server URL.

Reviewer 2: ```suggestion
When you're ready to deploy to production, run [`npx expo export`](/more/expo-cli#exporting) to create the server bundle in the **dist** directory. This server can be tested locally with `npx expo serve` (available in Expo SDK 52 and higher), visit the URL in a web browser or create a native build with the `origin` set to the local server URL.
```

- Let's make the dist directory name bold and use directory over folder as per our [writing style guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#filenames-as-bold-text)

You can deploy the server for production using [EAS Hosting](/eas/hosting/get-started) or another third-party service.

<BoxLink
  title="Deploy instantly with EAS"
  description="EAS Hosting is the best way to deploy your Expo API routes and servers."
  href="/eas/hosting/get-started"

>   Icon={Rocket02Icon}

Reviewer 2: ```suggestion
  Icon={Cloud01Icon}
```
